Wikipedia/Code/WelcomeAnalyticsAnimationView.swift

```diff
@@ -0,0 +1,239 @@
+        let leftTransform = CATransform3DMakeTranslation(-horizontalOffset, 0, 0)
+        let rightTransform = CATransform3DMakeTranslation(horizontalOffset, 0, 0)
+        let scaleZeroTransform = CATransform3DMakeScale(0, 0, 1)
+        let squashedHeightTransform = CATransform3DMakeScale(1, 0, 1)
+        let linesLeftTransform: CATransform3D  = CATransform3DConcat(scaleZeroTransform, rightTransform)
+        let linesRightTransform: CATransform3D  = CATransform3DConcat(scaleZeroTransform, leftTransform)
+
+        fileImgView.frame = self.bounds
+        fileImgView.image = UIImage(named: "ftux-file")
+        fileImgView.contentMode = UIViewContentMode.ScaleAspectFit;
+        fileImgView.layer.zPosition = 101
+        fileImgView.layer.opacity = 0
+        fileImgView.layer.transform = leftTransform;
+        self.addSubview(fileImgView)
+        
+        // Reminder: no need to start bars with opacity 0 because fileImgView, to which the bars are added, already
+        // fades from 0 to 1 opacity
+        bar1 = WelcomeBarShapeLayer(
+            unitRect: CGRectMake(0.313, 0.64, 0.039, 0.18),
+            referenceSize:self.frame.size,
+            transform: squashedHeightTransform
+        )
+        fileImgView.layer.addSublayer(bar1!)
+
+        bar2 = WelcomeBarShapeLayer(
+            unitRect: CGRectMake(0.383, 0.64, 0.039, 0.23),
+            referenceSize:self.frame.size,
+            transform: squashedHeightTransform
+        )
+        fileImgView.layer.addSublayer(bar2!)
+
+        bar3 = WelcomeBarShapeLayer(
+            unitRect: CGRectMake(0.453, 0.64, 0.039, 0.06),
+            referenceSize:self.frame.size,
+            transform: squashedHeightTransform
+        )
+        fileImgView.layer.addSublayer(bar3!)
+
+        bar4 = WelcomeBarShapeLayer(
+            unitRect: CGRectMake(0.523, 0.64, 0.039, 0.12),
+            referenceSize:self.frame.size,
+            transform: squashedHeightTransform
+        )
+        fileImgView.layer.addSublayer(bar4!)
+
+        bar5 = WelcomeBarShapeLayer(
+            unitRect: CGRectMake(0.593, 0.64, 0.039, 0.15),
+            referenceSize:self.frame.size,
+            transform: squashedHeightTransform
+        )
+        fileImgView.layer.addSublayer(bar5!)
+
+        let blue = UIColor(red: 0.1216, green: 0.5804, blue: 0.8667, alpha: 1.0).CGColor
+        let green = UIColor(red: 0.0000, green: 0.6941, blue: 0.5490, alpha: 1.0).CGColor
+        bar1?.fillColor = blue
+        bar2?.fillColor = green
+        bar3?.fillColor = blue
+        bar4?.fillColor = green
+        bar5?.fillColor = blue
+        
+        self.solidCircle = WelcomeCircleShapeLayer(
+            unitRadius: 0.235,
+            unitOrigin: CGPointMake(0.654, 0.41),
+            referenceSize: self.frame.size,
+            isDashed: false,
+            transform: scaleZeroTransform,
+            opacity:0.0
+        )
+        self.layer.addSublayer(self.solidCircle!)
```

Comment:
We should try to think of ways to get rid of the `!` when we can. Just because that is dangerous in swift land. 

One idea is to not make `self.solidCircle` an optional - you could do that by making it lazy so it always returns a value.